app/src/main/java/org/onionshare/android/tor/TorManager.kt

```diff
@@ -231,4 +295,100 @@ class TorManager @Inject constructor(
         return File(dataDir, "ControlSocket").absolutePath
     }
 
+    private suspend fun checkStartupProgress() {
+        LOG.info("Starting check job")
+
+        var usedBuiltInBridges = false
+        runWhenMoatTimeIsUp {
+            val bridges = try {
+                getBridgesFromMoat()
+            } catch (e: IOException) {
+                LOG.warn("Error getting bridges from moat: ", e)
+                null
+            }
+            if (bridges.isNullOrEmpty()) {
+                // fallback to built-in bridges if moat doesn't return anything
```

Comment:
Would be good to have some logging so we can follow the process.

If Moat doesn't return anything then it thinks Tor is unblocked in our location. We're trying built-in bridges here because we already tried without bridges and it didn't work, is that right?

Reply:
Right, trying without bridges is always first so far, but I guess we could also force certain bridges based on the location we determine.